### PR TITLE
Populate CAgg bucket catalog table for all CAggs

### PR DESCRIPTION
**Pull request by jnidzwetzki**
This changes the behavior of the CAgg catalog tables. From now on, all CAggs that use a time_bucket function create an entry in the catalog table continuous_aggs_bucket_function. In addition, the duplicate bucket_width attribute is removed from the catalog table continuous_agg.

--
Disable-check: force-changelog-file

### CODE REVIEW

File: sql/updates/reverse-dev.sql [hunk 1]
```diff
@@ -1,5 +1,140 @@
...
+        mat_hypertable_id;
+
+-- Migrate CAggs with fixed bucket on interval back
+WITH fixed_buckets AS (
+    SELECT * FROM _timescaledb_catalog.continuous_aggs_bucket_function WHERE bucket_fixed_width = true AND bucket_func::text LIKE '%time_bucket(interval%'
+)
+UPDATE _timescaledb_catalog._tmp_continuous_agg cagg SET bucket_width = _timescaledb_functions.interval_to_usec(
+      (SELECT bucket_width::interval FROM fixed_buckets fb WHERE cagg.mat_hypertable_id = fb.mat_hypertable_id)
+    )
+    WHERE mat_hypertable_id IN (SELECT mat_hypertable_id FROM fixed_buckets);
```
Comment (fabriziomello) [inline sql/updates/reverse-dev.sql:72]:
```suggestion
UPDATE _timescaledb_catalog._tmp_continuous_agg cagg
SET bucket_width = _timescaledb_functions.interval_to_usec(fb.bucket_width::interval)
FROM fixed_buckets fb
WHERE cagg.mat_hypertable_id = fb.mat_hypertable_id
```
IMHO using implicit joins is more readable (the same applies to the update below).

File: src/ts_catalog/continuous_agg.c [hunk 2]
```diff
@@ -401,6 +395,89 @@ continuous_agg_formdata_fill(FormData_continuous_agg *fd, const TupleInfo *ti)
 		heap_freetuple(tuple);
 }
 
+/*
+ * Fill the fields of a integer based bucketing function
+ */
+static void
+continuous_agg_fill_bucket_function_ib(ContinuousAggsBucketFunction *bf, bool *isnull,
```
Comment (fabriziomello) [inline src/ts_catalog/continuous_agg.c:402]:
```suggestion
cagg_fill_bucket_function_integer_based(ContinuousAggsBucketFunction *bf, bool *isnull,
```
Nit.

File: tsl/test/sql/exp_cagg_monthly.sql [hunk 3]
```diff
@@ -66,10 +66,6 @@ FROM _timescaledb_catalog.continuous_agg
 WHERE user_view_name = 'conditions_summary'
 \gset
 
-SELECT bucket_width
-FROM _timescaledb_catalog.continuous_agg
-WHERE mat_hypertable_id = :cagg_id;
-
```
Comment (fabriziomello) [inline tsl/test/sql/exp_cagg_monthly.sql:72]: Why remove it and not expand the query JOINING with the continuous_agg_bucket_function to show the bucket width?

Same applies to other two queries removed from exp_cagg_timezone.sql

Comment (jnidzwetzki) [reply tsl/test/sql/exp_cagg_monthly.sql:72]: This is tested in the next line. In this line, `continuous_aggs_bucket_function` is queried for the `cagg_id`. The same is true for the removed lines in `exp_cagg_timezone`, `continuous_aggs_bucket_function.bucket_width` is read in the next lines.

File: src/ts_catalog/continuous_agg.c (hunk 2 above)
Comment (jnidzwetzki) [reply src/ts_catalog/continuous_agg.c]: Naming of these two functions is changed.

File: sql/updates/reverse-dev.sql [hunk 4]
```diff
@@ -1,5 +1,141 @@
...
+DROP PROCEDURE IF EXISTS _timescaledb_internal.cagg_migrate_execute_enable_policies (_timescaledb_catalog.continuous_agg, _timescaledb_catalog.continuous_agg_migrate_plan_step);
+DROP PROCEDURE IF EXISTS _timescaledb_functions.cagg_migrate_execute_enable_policies (_timescaledb_catalog.continuous_agg, _timescaledb_catalog.continuous_agg_migrate_plan_step);
+DROP PROCEDURE IF EXISTS _timescaledb_internal.cagg_migrate_execute_copy_policies (_timescaledb_catalog.continuous_agg, _timescaledb_catalog.continuous_agg_migrate_plan_step);
+DROP PROCEDURE IF EXISTS _timescaledb_functions.cagg_migrate_execute_copy_policies (_timescaledb_catalog.continuous_agg, _timescaledb_catalog.continuous_agg_migrate_plan_step);
+DROP PROCEDURE IF EXISTS _timescaledb_internal.cagg_migrate_execute_refresh_new_cagg (_timescaledb_catalog.continuous_agg, _timescaledb_catalog.continuous_agg_migrate_plan_step);
+DROP PROCEDURE IF EXISTS _timescaledb_functions.cagg_migrate_execute_refresh_new_cagg (_timescaledb_catalog.continuous_agg, _timescaledb_catalog.continuous_agg_migrate_plan_step);
```
Comment (akuzm) [inline sql/updates/reverse-dev.sql:29]: Why do we have to drop these procedures? I see that they are dropped in both directions, are they created somewhere else?

**Review by akuzm — APPROVED** [review]

**Review by fabriziomello — APPROVED** [review]

**Pull request merged**